framework/src/executioners/Transient.C

```diff
@@ -160,7 +160,7 @@ Transient::Transient(const InputParameters & parameters)
     _time_interval(declareRecoverableData<bool>("time_interval", false)),
     _start_time(getParam<Real>("start_time")),
     _timestep_tolerance(getParam<Real>("timestep_tolerance")),
-    _target_time(declareRecoverableData<Real>("target_time", -1)),
+    _target_time(declareRecoverableData<Real>("target_time", -std::numeric_limits<Real>::max())),
```

Comment:
If we do not set start_time in the input file, I will have a negative `start_time`?

Reply:
This PR does not change the behavior w.r.t `start_time`. For the `_target_time`, it was currently assumed that if it is negative, then it has not been set. This logic does not work if we are starting the simulation at negative times, in which case the `_target_time` could have been set but still be negative.